File: 961-long-pressed-name/long-pressed-name.java
class Solution {
    public boolean isLongPressedName(String name, String typed) {
        int i = 0; // name
        int j = 0; // typed

        while(i < name.length() && j < typed.length()){
            if(name.charAt(i) == typed.charAt(j)){
                i++;
                j++;
            }
            else if(j-1>=0 && typed.charAt(j) == typed.charAt(j - 1)){
                j++;
            }
            else{
                return false;
            }
        }

        while(i < name.length()){
            if(j<typed.length() && name.charAt(i) == typed.charAt(j)){
                i++;
                j++;
            }
            else{
                return false;
            }
        }

        while(j < typed.length()){
            if(j-1>=0 && typed.charAt(j) == typed.charAt(j - 1)){
                j++;
            }
            else{
                return false;
            }
        }

        return true;
        

    }

    /**

        alexd
           i
           j
        ale
         
     */
}

File: 961-long-pressed-name/long-pressed-name.py
class Solution:
    def isLongPressedName(self, name: str, typed: str) -> bool:
        i = 0
        j = 0

        while i < len(name) and j < len(typed):
            if name[i] == typed[j]:
                i = i + 1
                j = j + 1
            elif j-1>=0 and typed[j] == typed[j - 1]:
                j = j + 1
            else:
                return False

        if j == len(typed) and i < len(name):
            return False

        while j < len(typed):
            if j-1>=0 and typed[j] == typed[j - 1]:
                j = j + 1
            else:
                return False

        return True
